content/general/events.textile

```diff
@@ -0,0 +1,374 @@
+
+  h5(#failures). Failures and back off
+
+  * If the endpoint for any of the WebHook requests respond with an HTTP status code that does not indicate success i.e. @200 - 209@, then Ably will retry that failed request
+  * Every retry is performed with an incrementing back off that is calculated as @delay = delay * sqrt(2)@ where delay is initially @1@. For example, if the initial WebHook request fails, and subsequent for retries fail, the back off delays for each request would look as follows: @initial request > wait 1.4s > 1st retry > wait 2s > 2nd retry > wait 2.8s > 3rd retry > wait 4s > 4th retry > wait 5.6s > successful request@
+  * The back off for consecutively failing requests will increase until it reaches 60s. All subsequent retries for failed requests will then be made every 60s until a request is successful
+  * The queue of events is retained for 5 minutes. If an event cannot be delivered within 5 minutes, then the events are discarded to prevent the queue from growing indefinitely
+
+h3(#envelope). Envelopes
+
+When you configure a Reactor Event rule using "single requests":#batching, you are given the option to envelope messages, which is enabled by default. In most cases, we believe an enveloped message provides more flexibility as it contains additional metadata in a portable format that can be useful such as the @clientId@ of the publisher, or the @channel@ name the message originated from.
+
+However, where performance is a primary concern, you may choose not to envelope messages and instead have only the message payload (@data@ element) published. This has the advantage of requiring one less parsing step, however decoding of the raw payload in the published message will be your responsibility.
+
+Note that messages published to queues are by default encoded as JSON (a text format), however you can choose to have messages encoded with "MsgPack":msgpack.org (a binary format) in your Reactor rules.
+
+Check out examples of "enveloped":#envelope-examples and "non-enveloped":#no-envelope-examples examples down below.
+
+h3(#encoding). Payload Encoding
+
+The encoding of payloads sent is defined when setting up a Reactor Rule in the "Reactor tab of your app":https://support.ably.io/solution/articles/3000074406. You can have the message sent in JSON format, or as a "MessagePack":https://msgpack.org payload.
+
+* "JSON":https://www.json.org (JavaScript Object Notation): An efficient data-interchange format which is fairly standard and provides simple text based encoding.
+* "MessagePack":https://msgpack.org: An efficient binary serialization format that is similar to JSON, but faster and smaller.
```

Comment:
Not necessary faster. But will be smaller. Worth highlighting that it's most useful for binary payloads, because if you have a json envelope with a binary payload, the payload will need to be base64-encoded.